src/test/java/io/stargate/sgv2/jsonapi/api/v1/UpdateManyIntegrationTest.java

```diff
@@ -0,0 +1,342 @@
+                                }
+                                """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.matchedCount", is(20))
+          .body("status.modifiedCount", is(20))
+          .body("status.moreData", nullValue());
+
+      json =
+          """
+                        {
+                          "find": {
+                            "filter" : {"active_user": false}
+                          }
+                        }
+                        """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.count", is(20));
+    }
+
+    @Test
+    @Order(5)
+    public void deleteManyLimitMoreDataFlag() {
+      insert(25);
+      String json =
+          """
+                                {
+                                  "updateMany": {
+                                    "filter" : {"active_user" : true},
+                                    "update" : {"$set" : {"active_user": false}}
+                                  }
+                                }
+                                """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.matchedCount", is(20))
+          .body("status.modifiedCount", is(20))
+          .body("status.moreData", is(true));
+
+      json =
+          """
+                        {
+                          "find": {
+                            "filter" : {"active_user": true}
+                          }
+                        }
+                        """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.count", is(5));
+    }
+
+    @Test
+    @Order(6)
+    public void updateManyUpsert() {
+      insert(5);
+      String json =
+          """
+                                    {
+                                      "updateMany": {
+                                        "filter" : {"_id": "doc6"},
```

Comment:
Another detail worth confirming: do additional fields in the filter get added to the resulting document on upsert? For example, if this is `"filter" : {"_id": "doc6", "answer": 42}`, does the resulting document have an `answer` property?

Reply:
Good question: I think method "getEmptyDocument(s)" (or renamed as "getNewDocument") was only inserting document id?

Reply:
@vkarpov15 During upsert the getNewDocument() method creates document populated with _id if it is part of the  filter condition. So the resulting document will not have the answer property populated based on the filter condition. I will add a test case for this in a separate PR.

Reply:
@maheshrajamani this is a somewhat important detail. Upserting should apply any fields from the filter that don't have a non- `$eq` query operator. Below is some output from the mongodb shell that shows this behavior.

Mongoose does use this behavior for setting schema defaults on upsert.

```
MongoDB Enterprise > db.tests.countDocuments({})
0
MongoDB Enterprise > db.tests.updateOne({ _id: 'foo', answer: 42 }, { $set: { bar: 'baz' } }, { upsert: true })
{
	"acknowledged" : true,
	"matchedCount" : 0,
	"modifiedCount" : 0,
	"upsertedId" : "foo"
}
MongoDB Enterprise > db.tests.findOne()
{ "_id" : "foo", "answer" : 42, "bar" : "baz" }
MongoDB Enterprise > 
MongoDB Enterprise > db.tests.updateOne({ _id: 'foo', answer: { $eq: 42 }, count: { $gte: 42 } }, { $set: { bar: 'baz' } }, { upsert: true })
{
	"acknowledged" : true,
	"matchedCount" : 0,
	"modifiedCount" : 0,
	"upsertedId" : "foo"
}
MongoDB Enterprise > db.tests.findOne()
{ "_id" : "foo", "answer" : 42, "bar" : "baz" }
MongoDB Enterprise > 

```

Reply:
@vkarpov15 Got it. Will add equals condition filter fields to the newly created document and will submit a new PR.

Reply:
Resolved in PR #200